src/scitype.jl

```diff
@@ -72,78 +33,128 @@ In general, one cannot infer the scitype of an object of type
+Fallback_Scitype(::Type, C) = Unknown
 
-# to distinguish between Any type and Union{T,Missing} for some more
-# specialised `T`, we define the Any case explicitly
-Scitype(::Type{Any}, ::Convention) = Unknown
+# To distinguish between `Any` type and `Union{T, Missing}` for some more
+# specialised `T`, we define the `Any` case explicitly.
+Fallback_Scitype(::Type{Any}, C) = Unknown
 
-# for the case Union{T,Missing} we return Union{S,Missing} with S
-# the scientific type corresponding to T
-Scitype(::Type{Union{T,Missing}}, C::Convention) where T =
-    Union{Missing,Scitype(T, C)}
+# For the case `Union{T, Missing}` we return `Union{S, Missing}` with `S`
+# the scientific type corresponding to `T`.
+function Fallback_Scitype(::Type{Union{T, Missing}}, C) where T
+    return Union{Scitype(Missing, C), Scitype(T, C)}
+end
 
-# for the case Missing, we return Missing
-Scitype(::Type{Missing}, C::Convention) = Missing
+# For the case `Missing` and `Nothing`, 
+# we return `Missing` and `Nothing` respectively.
+Fallback_Scitype(::Type{Missing}, C) = Missing
+Fallback_Scitype(::Type{Nothing}, C) = Nothing
 
-Scitype(::Type{Nothing}, C::Convention) = Nothing
+# ---------------------------------------------------------------------------------------
+# Fallbacks for `scitype` of tuples and abstractarrays.
 
-# Broadcasting over arrays
+function fallback_scitype(@nospecialize(t::Tuple), C; kw...)
+    if length(t) <= TUPLE_SPECIALIZATION_THRESHOLD
+        return _tuple_scitype(t, C; kw...)
+    else
+        return Tuple{(scitype(el, C; kw...) for el in t)...}
+    end
+end
 
-scitype(A::Arr{T}, C::Convention, ::Val{:other}; kw...) where T =
-    arr_scitype(A, C, Scitype(T, C); kw...)
+@inline function _tuple_scitype(t::T, C; kw...) where {T}
+    if @generated
+        stypes = Tuple(
+            :(scitype(getindex(t, $i), C; kw...)) for i in 1:fieldcount(T)
+        )
+        return :(Tuple{$(stypes...)})
+    else
+        stypes_ = Tuple(
+            scitype(getindex(t, i), C; kw...) for i in 1:fieldcount(T)
+        )
+        return Tuple{(stypes_)...}
+    end
+end
+
+function fallback_scitype(A::Arr{T}, C; kw...) where T
+    return arr_scitype(A, Scitype(T, C), C; kw...)
+end
 
 """
-    arr_scitype(A, C, S; tight)
+    arr_scitype(A, S, C; tight)
 
-Return the scitype associated with an  array `A` of type `{T,N}` assuming an
-explicit `Scitype` correspondance exist mapping `T` to `S`.
+Return the scitype associated with an  array `A` of type `{T, N}` assuming an
+explicit `Scitype` correspondence exist mapping `T` to `S`.
 If `tight=true` and `T>:Missing` then the function checks whether there are
-"true missing values", otherwise it constructs a "tight copy" of the array
-without a `Union{Missing,S}` type.
+"true missing values", otherwise it returns the scitype asscoiated with a 
+"tight copy" of the array.
+**Note**
+The eltype of a "tight_copy" of an array with eltype `Union{T, Missing}` is `T`  
```

Comment:
I'm a bit confused by the last "Note". Won't the eltype depend on where or not there are indeed `missing` elements? You seem to suggest it will always be `T`.

Reply:
Am saying that when an array has an `eltype` of `Union{T, Missing}`, then the `eltype` of the corresponding tight_copy of the array will have an `eltype` `T`.